src/libraries/BytesLib.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```

Comment:
These functions have been copied in from the UniversalRouter. We cant import it from there otherwise we're going to have a circular dependency between V4-periphery and UR lol. I know they look ugly but theyre audited and live in UR

Reply:
Yeah we should really be importing :/ but maybe we can just make this library our own then? Cause I find these really hard to read.. when you do .toUint256Array(0) it seems like its initializing a 0 length uint256 array.

& its also unnecessarily operating on the full returned calldata twice?

I'd rather the syntax just be (uint256[] arr, bytes b) = data.decode(); 

cause we dont use `toUint256Array` or `toBytesArray` by themselves afaik

Reply:
yeah nice idea, i've made a single function that does it all. Optimised for this usecase. Saves like another 300 gas woo